pkg/controller/job/job_controller_test.go

```diff
@@ -725,175 +717,163 @@ func TestControllerSyncJob(t *testing.T) {
 	}
 
 	for name, tc := range testCases {
-		for _, wFinalizers := range []bool{false, true} {
-			t.Run(fmt.Sprintf("%s, finalizers=%t", name, wFinalizers), func(t *testing.T) {
-				if wFinalizers && tc.podControllerError != nil {
-					t.Skip("Can't track status if finalizers can't be removed")
-				}
-				if tc.wFinalizersExclusive != nil && *tc.wFinalizersExclusive != wFinalizers {
-					t.Skipf("Test is exclusive for wFinalizers=%t", *tc.wFinalizersExclusive)
-				}
-				defer featuregatetesting.SetFeatureGateDuringTest(t, feature.DefaultFeatureGate, features.JobReadyPods, tc.jobReadyPodsEnabled)()
-
-				// job manager setup
-				clientSet := clientset.NewForConfigOrDie(&restclient.Config{Host: "", ContentConfig: restclient.ContentConfig{GroupVersion: &schema.GroupVersion{Group: "", Version: "v1"}}})
-				manager, sharedInformerFactory := newControllerFromClient(clientSet, controller.NoResyncPeriodFunc)
-				fakePodControl := controller.FakePodControl{Err: tc.podControllerError, CreateLimit: tc.podLimit}
-				manager.podControl = &fakePodControl
-				manager.podStoreSynced = alwaysReady
-				manager.jobStoreSynced = alwaysReady
-
-				// job & pods setup
-				job := newJob(tc.parallelism, tc.completions, tc.backoffLimit, tc.completionMode)
-				job.Spec.Suspend = pointer.Bool(tc.suspend)
-				key, err := controller.KeyFunc(job)
-				if err != nil {
-					t.Errorf("Unexpected error getting job key: %v", err)
-				}
-				if tc.fakeExpectationAtCreation < 0 {
-					manager.expectations.ExpectDeletions(key, int(-tc.fakeExpectationAtCreation))
-				} else if tc.fakeExpectationAtCreation > 0 {
-					manager.expectations.ExpectCreations(key, int(tc.fakeExpectationAtCreation))
-				}
-				if tc.wasSuspended {
-					job.Status.Conditions = append(job.Status.Conditions, *newCondition(batch.JobSuspended, v1.ConditionTrue, "JobSuspended", "Job suspended", realClock.Now()))
-				}
-				if wFinalizers {
-					job.Annotations = map[string]string{
-						batch.JobTrackingFinalizer: "",
-					}
-				}
-				if tc.deleting {
-					now := metav1.Now()
-					job.DeletionTimestamp = &now
-				}
-				sharedInformerFactory.Batch().V1().Jobs().Informer().GetIndexer().Add(job)
-				podIndexer := sharedInformerFactory.Core().V1().Pods().Informer().GetIndexer()
-				setPodsStatuses(podIndexer, job, tc.pendingPods, tc.activePods, tc.succeededPods, tc.failedPods, tc.readyPods)
-				setPodsStatusesWithIndexes(podIndexer, job, tc.podsWithIndexes)
-
-				actual := job
-				manager.updateStatusHandler = func(ctx context.Context, job *batch.Job) (*batch.Job, error) {
-					actual = job
-					return job, nil
-				}
+		t.Run(name, func(t *testing.T) {
+			if tc.podControllerError != nil {
```

Comment:
Following up the previous discussion:

Update these tests in a follow up PR.

Reply:
I'll create a ticket for tracking (mostly for me! and whoever would be willing to take it).